### PR TITLE
Allow custom root PEM

### PR DESCRIPTION
**Pull request by lukehinds**
This change allows use of a custom env `COSIGN_ROOT` for those
who wish to use their own root CA in an off public net fulcio
instance.

Ultimately we might want to do this with a flag, but it requires
a fair amount of rewrite, so this seems a good pragamtic bridge
without being to disruptive prior to 1.0

Resolves: #87 

Signed-off-by: Luke Hinds <lhinds@redhat.com>

### CODE REVIEW

File: pkg/cosign/fulcio/fulcio.go [hunk 1]
```diff
@@ -46,6 +47,7 @@ const (
 	FlowNormal = "normal"
 	FlowDevice = "device"
 	FlowToken  = "token"
+	altRoot    = "COSIGN_ROOT"
```
Comment (dlorenc) [inline pkg/cosign/fulcio/fulcio.go:50]: nit: maybe "SIGSTORE_ROOT_DIR" instead? These are really the roots for the sigstore project and aren't cosign specific.

**Review by dlorenc — COMMENTED** [review]
One nit on the env var name. We should also get this into the README.md somewhere.

File: pkg/cosign/fulcio/fulcio.go (hunk 1 above)
Comment (lukehinds) [reply pkg/cosign/fulcio/fulcio.go:50]: agree on both, I will add the other flags added too.

File: pkg/cosign/fulcio/fulcio.go [hunk 2]
```diff
@@ -198,7 +200,16 @@ var Roots *x509.CertPool
 
 func init() {
 	cp := x509.NewCertPool()
-	if !cp.AppendCertsFromPEM([]byte(rootPem)) {
+	rootEnv := os.Getenv(altRoot)
+	if rootEnv != "" {
+		raw, err := ioutil.ReadFile(rootEnv)
```
Comment (dlorenc) [inline pkg/cosign/fulcio/fulcio.go:205]: Oh sorry - I just realized this is a file, not a directory. I think we have two options:

- Keep it as a directory and concatenate all the pem files in it using something like filepath.Glob (this is pretty standard for roots, you can easily have multiple)
- Rename it to "_FILE" and leave it this way. 

My hunch in the long term is that we'll want the ability to have multiple (adding yours in addition to the public good one)

**Review by dlorenc — APPROVED** [review]

**Pull request merged**